pkg/util/webhookpatch_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+func TestWebhookPatch(t *testing.T) {
+	ts := []struct {
+		configs     admissionregistrationv1beta1.MutatingWebhookConfigurationList
+		configName  string
+		webhookName string
+		pemData     []byte
+		expected    string
+	}{
+		{
+			admissionregistrationv1beta1.MutatingWebhookConfigurationList{},
+			"config1",
+			"webhook1",
+			[]byte("fake CA"),
+			"mutatingwebhookconfigurations.admissionregistration.k8s.io \"config1\" not found",
```

Comment:
Testing specific error message strings tends to be fragile. Is presence of error sufficient, or does this specific error type matter?

Reply:
I think we'd better be strict on checking every specific error. Ignoring the type of error could hide some problems in practice.

Reply:
Strict checking is fine. Checking the actually error type and not the string message would be better, e.g.

https://github.com/istio/istio/blob/master/mixer/pkg/config/crd/admit.go#L294
```golang
		serr, ok := err.(*apierrors.StatusError)
		if !ok || serr.ErrStatus.Code != http.StatusNotFound {
			log.Warnf("Could not delete previously created AdmissionRegistration: %v", err)
		}
```